x-pack/plugins/fleet/server/saved_objects/index.ts

```diff
@@ -143,11 +144,26 @@ const getSavedObjectTypes = (
         is_preconfigured: { type: 'boolean', index: false },
         ssl: { type: 'binary' },
         proxy_id: { type: 'keyword' },
+        shipper: {
+          properties: {
+            disk_queue_enabled: { type: 'boolean', index: false },
+            disk_queue_path: { type: 'keyword', index: false },
+            disk_queue_max_size: { type: 'integer', index: false },
+            disk_queue_encryption_enabled: { type: 'boolean', index: false },
+            disk_queue_compression_enabled: { type: 'boolean', index: false },
+            compression_level: { type: 'integer', index: false },
+            loadbalance: { type: 'boolean' },
+            mem_queue_events: { type: 'integer', index: false },
+            queue_flush_timeout: { type: 'integer', index: false },
+            max_batch_bytes: { type: 'integer', index: false },
```

Comment:
Some of these parameters could still change definition in the finalised shipper (see [this discussion](https://github.com/elastic/kibana/issues/141508#issuecomment-1307665963)). 

I'm not sure what's the best course of action here, can we change them later if needed or is it better to remove them from this PR (that would mean removing the UI related part as well) and add them back in a follow up PR?

Reply:
At the moment, as long as the additional changes are performed before FF, this is fine-ish (even if ideally only the definitive version should be merged into main).  

Now, about the future: we should be communicating on it soon, but it is extremely likely that at some point not that far in time, mappings removals will no longer be accepted, meaning that once a field definition has been pushed to production (meaning shipped in a release), it will have to stay here (and be maintained) permanently, so we *strongly* encourage teams to stop pushing mapping / schema changes they're not fully confident about.